File: src/09.tree/42.700.go
package tree

func SearchBST(root *TreeNode, val int) *TreeNode {
	if root == nil {
		return nil
	}

	if val > root.Val {
		return SearchBST(root.Right, val)
	} else if val < root.Val {
		return SearchBST(root.Left, val)
	}

	return root
}

File: src/09.tree/42.700.ts
import { TreeNode } from "./01.100";

export function searchBST(root: TreeNode | null, val: number): TreeNode | null {
    if (!root) {
        return null;
    }

    if (val > root.val) {
        return searchBST(root.right, val);
    } else if (val < root.val) {
        return searchBST(root.left, val);
    }

    return root;
}
